backend/pom.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Amazon.com Inc or its affiliates and the project contributors
+// Written by James Shubin <purple@amazon.com> and the project contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+// We will never require a CLA to submit a patch. All contributions follow the
+// `inbound == outbound` rule.
+//
+// This is not an official Amazon product. Amazon does not offer support for
+// this project.
+
+// TODO: should this be a subpackage?
+package backend
+
+import (
+	"context"
+	"sort"
+	"encoding/xml"
+
+	"github.com/awslabs/yesiscan/interfaces"
+	"github.com/awslabs/yesiscan/util/errwrap"
+	"github.com/awslabs/yesiscan/util/licenses"
+)
+
+const (
+	// PomFilename is the file name used by the pom
+	// files.
+	POMFilename = "pom.xml"
+
+)
+
+//POM or pom.xml files are created from running Java files and they
+//sometimes contain the license of the project.We get the license id(s)
+//by parsing the pom.xml file for its license name(s).
+type Pom struct {
+	Debug bool
+	Logf  func(format string, v ...interface{})
+}
+
+
+type LicenseNames struct{
+	Names []string `xml:"licenses>license>name"`
+}
+
+func (obj *Pom) String() string {
+	return "pom"
+}
+
+func (obj *Pom) ScanData(ctx context.Context, data []byte, info *interfaces.Info) (*interfaces.Result, error) {
+	//this check is taking place with the assumption that 
+	//the file that will be scanned will have to be named "pom.xml"
+	if info.FileInfo.Name() != POMFilename {
+		return nil, nil // skip
+	}
+	if info.FileInfo.IsDir() {
+		return nil, nil // skip
+	}
+	if len(data) == 0 {
+		return nil, nil // skip
+	}
+
+	licenseMap := make(map[string]struct{})
+
+	var list LicenseNames
+	//parsing pom.xml file
+	err := xml.Unmarshal(data, &list)
+	if  err != nil {
+		return nil, errwrap.Wrapf(err, "parse error")
+	}
+	
+	//lid is license id
+	for _, lid := range list.Names {
+		licenseMap[lid] = struct{}{}
+	}
+
+	if len(licenseMap) == 0 {
+		// NOTE: this is NOT the same as interfaces.ErrUnknownLicense
```

Comment:
Do you understand this comment ? Please explain it so I know you didn't just blatantly copy-paste it ;) To be clear, you are encouraged to blatantly copy and paste this kind of reasoning, but you must also make sure you are sure you understand it.